src/config.js

```diff
@@ -133,29 +149,20 @@ export default class Config {
         );
     }
     switch (true) {
-      case !!this.inputs.token && !!this.inputs.webhook:
-        core.debug(
-          "Setting the provided token and webhook as secret variables.",
-        );
-        core.setSecret(this.inputs.token);
-        core.setSecret(this.inputs.webhook);
+      case !!core.getInput("token") && !!core.getInput("webhook"):
         throw new SlackError(
           core,
           "Invalid input! Either the token or webhook is required - not both.",
         );
-      case !!this.inputs.token:
-        core.debug("Setting the provided token as a secret variable.");
-        core.setSecret(this.inputs.token);
-        if (!this.inputs.method) {
+      case !!this.inputs.method:
+        if (!this.inputs.token) {
```

Comment:
Special care was taken in flipping these cases with wording since we can't assume a set `this.inputs.token` means that technique is being used.

Instead, I think we must check for a `method` instead of `token`, and always check `method` before `webhook` 🤔